ruby 1.9.3p194 (2012-04-20 revision 35410) [x86_64-linux]
Run options: 

# Running tests:

.....F......EEF..F.....$ ruby  script/work_flow.rb --merge-down
@errors="/home/greg/Desktop/src/Open-Table-Explorer/app/models/work_flow.rb:73:in `merge_range': :work_flow not found in [:passed, :testing, :edited] or {-2=>:\"origin/master\", -1=>:master} (RuntimeError)\n\tfrom /home/greg/Desktop/src/Open-Table-Explorer/app/models/work_flow.rb:289:in `merge_down'\n\tfrom script/work_flow.rb:114:in `block in <main>'\n\tfrom script/work_flow.rb:97:in `each'\n\tfrom script/work_flow.rb:97:in `<main>'\n"
@process_status=#<Process::Status: pid 19516 exit 1>
$ echo "Hello World"
Hello World
$ echo "1 2;3 4"
1 2;3 4
$ /sbin/ifconfig
eth0      Link encap:Ethernet  HWaddr f8:0f:41:d2:26:c7  
          inet addr:172.31.41.166  Bcast:172.31.41.255  Mask:255.255.255.0
          inet6 addr: fe80::fa0f:41ff:fed2:26c7/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:1761595 errors:0 dropped:0 overruns:0 frame:0
          TX packets:1494023 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:1166773469 (1.0 GiB)  TX bytes:202909977 (193.5 MiB)

lo        Link encap:Local Loopback  
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:273143 errors:0 dropped:0 overruns:0 frame:0
          TX packets:273143 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0 
          RX bytes:88017426 (83.9 MiB)  TX bytes:88017426 (83.9 MiB)

wlan0     Link encap:Ethernet  HWaddr 30:10:b3:40:00:e5  
          inet addr:192.168.1.221  Bcast:192.168.1.255  Mask:255.255.255.0
          inet6 addr: 2605:e000:5b0a:3e00:147a:fef:938:75a8/64 Scope:Global
          inet6 addr: fde9:7883:b343:0:3210:b3ff:fe40:e5/64 Scope:Global
          inet6 addr: fe80::3210:b3ff:fe40:e5/64 Scope:Link
          inet6 addr: fde9:7883:b343:0:147a:fef:938:75a8/64 Scope:Global
          inet6 addr: 2605:e000:5b0a:3e00:3210:b3ff:fe40:e5/64 Scope:Global
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:911018 errors:0 dropped:3 overruns:0 frame:0
          TX packets:318424 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:174958035 (166.8 MiB)  TX bytes:39902983 (38.0 MiB)

$ file -b script/work_flow.rb
Ruby script, ASCII text executable
$ file -b --mime script/work_flow.rb
text/x-ruby; charset=us-ascii
work_flow --<command> <file>
F........E

Finished tests in 3.913534s, 8.4323 tests/s, 168.6455 assertions/s.

  1) Failure:
test_bracketing_versions?(WorkFlowTest) [test/unit/work_flow_test.rb:173]:
<-2> expected but was
<-1>.

  2) Error:
test_deserve_command(WorkFlowTest):
NoMethodError: undefined method `+' for nil:NilClass
    /home/greg/Desktop/src/Open-Table-Explorer/app/models/repository.rb:145:in `ruby_test_string'
    /home/greg/Desktop/src/Open-Table-Explorer/app/models/repository.rb:154:in `error_score?'
    test/unit/work_flow_test.rb:255:in `test_deserve_command'
    /usr/lib/ruby/vendor_ruby/minitest/unit.rb:1058:in `run'
    /usr/lib/ruby/1.9.1/test/unit/testcase.rb:17:in `run'
    /usr/lib/ruby/vendor_ruby/active_support/testing/setup_and_teardown.rb:36:in `block in run'
    /usr/lib/ruby/vendor_ruby/active_support/callbacks.rb:403:in `_run__1874944238495754569__setup__1279465477025375313__callbacks'
    /usr/lib/ruby/vendor_ruby/active_support/callbacks.rb:405:in `__run_callback'
    /usr/lib/ruby/vendor_ruby/active_support/callbacks.rb:385:in `_run_setup_callbacks'
    /usr/lib/ruby/vendor_ruby/active_support/callbacks.rb:81:in `run_callbacks'
    /usr/lib/ruby/vendor_ruby/active_support/testing/setup_and_teardown.rb:35:in `run'
    /usr/lib/ruby/vendor_ruby/minitest/unit.rb:823:in `block in _run_suite'
    /usr/lib/ruby/vendor_ruby/minitest/unit.rb:815:in `map'
    /usr/lib/ruby/vendor_ruby/minitest/unit.rb:815:in `_run_suite'
    /usr/lib/ruby/1.9.1/test/unit.rb:565:in `block in _run_suites'
    /usr/lib/ruby/1.9.1/test/unit.rb:563:in `each'
    /usr/lib/ruby/1.9.1/test/unit.rb:563:in `_run_suites'
    /usr/lib/ruby/vendor_ruby/minitest/unit.rb:775:in `_run_anything'
    /usr/lib/ruby/vendor_ruby/minitest/unit.rb:964:in `run_tests'
    /usr/lib/ruby/vendor_ruby/minitest/unit.rb:951:in `block in _run'
    /usr/lib/ruby/vendor_ruby/minitest/unit.rb:950:in `each'
    /usr/lib/ruby/vendor_ruby/minitest/unit.rb:950:in `_run'
    /usr/lib/ruby/vendor_ruby/minitest/unit.rb:939:in `run'
    /usr/lib/ruby/1.9.1/test/unit.rb:21:in `run'
    /usr/lib/ruby/1.9.1/test/unit.rb:326:in `block (2 levels) in autorun'
    /usr/lib/ruby/1.9.1/test/unit.rb:27:in `run_once'
    /usr/lib/ruby/1.9.1/test/unit.rb:325:in `block in autorun'

  3) Error:
test_deserving_branch?(WorkFlowTest):
NoMethodError: undefined method `+' for nil:NilClass
    /home/greg/Desktop/src/Open-Table-Explorer/app/models/repository.rb:145:in `ruby_test_string'
    /home/greg/Desktop/src/Open-Table-Explorer/app/models/repository.rb:154:in `error_score?'
    test/unit/work_flow_test.rb:216:in `block in test_deserving_branch?'
    test/unit/work_flow_test.rb:214:in `each_pair'
    test/unit/work_flow_test.rb:214:in `test_deserving_branch?'
    /usr/lib/ruby/vendor_ruby/minitest/unit.rb:1058:in `run'
    /usr/lib/ruby/1.9.1/test/unit/testcase.rb:17:in `run'
    /usr/lib/ruby/vendor_ruby/active_support/testing/setup_and_teardown.rb:36:in `block in run'
    /usr/lib/ruby/vendor_ruby/active_support/callbacks.rb:403:in `_run__1874944238495754569__setup__1279465477025375313__callbacks'
    /usr/lib/ruby/vendor_ruby/active_support/callbacks.rb:405:in `__run_callback'
    /usr/lib/ruby/vendor_ruby/active_support/callbacks.rb:385:in `_run_setup_callbacks'
    /usr/lib/ruby/vendor_ruby/active_support/callbacks.rb:81:in `run_callbacks'
    /usr/lib/ruby/vendor_ruby/active_support/testing/setup_and_teardown.rb:35:in `run'
    /usr/lib/ruby/vendor_ruby/minitest/unit.rb:823:in `block in _run_suite'
    /usr/lib/ruby/vendor_ruby/minitest/unit.rb:815:in `map'
    /usr/lib/ruby/vendor_ruby/minitest/unit.rb:815:in `_run_suite'
    /usr/lib/ruby/1.9.1/test/unit.rb:565:in `block in _run_suites'
    /usr/lib/ruby/1.9.1/test/unit.rb:563:in `each'
    /usr/lib/ruby/1.9.1/test/unit.rb:563:in `_run_suites'
    /usr/lib/ruby/vendor_ruby/minitest/unit.rb:775:in `_run_anything'
    /usr/lib/ruby/vendor_ruby/minitest/unit.rb:964:in `run_tests'
    /usr/lib/ruby/vendor_ruby/minitest/unit.rb:951:in `block in _run'
    /usr/lib/ruby/vendor_ruby/minitest/unit.rb:950:in `each'
    /usr/lib/ruby/vendor_ruby/minitest/unit.rb:950:in `_run'
    /usr/lib/ruby/vendor_ruby/minitest/unit.rb:939:in `run'
    /usr/lib/ruby/1.9.1/test/unit.rb:21:in `run'
    /usr/lib/ruby/1.9.1/test/unit.rb:326:in `block (2 levels) in autorun'
    /usr/lib/ruby/1.9.1/test/unit.rb:27:in `run_once'
    /usr/lib/ruby/1.9.1/test/unit.rb:325:in `block in autorun'

  4) Failure:
test_diff_command?(WorkFlowTest) [test/unit/work_flow_test.rb:91]:
<nil> expected to not be nil.

  5) Failure:
test_goldilocks(WorkFlowTest) [test/unit/work_flow_test.rb:181]:
<nil> expected to not be nil.

  6) Failure:
test_merge_command(WorkFlowTest) [test/unit/work_flow_test.rb:247]:
self=$ ruby  script/work_flow.rb --merge-down
@errors="/home/greg/Desktop/src/Open-Table-Explorer/app/models/work_flow.rb:73:in `merge_range': :work_flow not found in [:passed, :testing, :edited] or {-2=>:\"origin/master\", -1=>:master} (RuntimeError)\n\tfrom /home/greg/Desktop/src/Open-Table-Explorer/app/models/work_flow.rb:289:in `merge_down'\n\tfrom script/work_flow.rb:114:in `block in <main>'\n\tfrom script/work_flow.rb:97:in `each'\n\tfrom script/work_flow.rb:97:in `<main>'\n"
@process_status=#<Process::Status: pid 19516 exit 1>
$ echo "Hello World"
Hello World
$ echo "1 2;3 4"
1 2;3 4
$ /sbin/ifconfig
eth0      Link encap:Ethernet  HWaddr f8:0f:41:d2:26:c7  
          inet addr:172.31.41.166  Bcast:172.31.41.255  Mask:255.255.255.0
          inet6 addr: fe80::fa0f:41ff:fed2:26c7/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:1761595 errors:0 dropped:0 overruns:0 frame:0
          TX packets:1494023 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:1166773469 (1.0 GiB)  TX bytes:202909977 (193.5 MiB)

lo        Link encap:Local Loopback  
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:273143 errors:0 dropped:0 overruns:0 frame:0
          TX packets:273143 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0 
          RX bytes:88017426 (83.9 MiB)  TX bytes:88017426 (83.9 MiB)

wlan0     Link encap:Ethernet  HWaddr 30:10:b3:40:00:e5  
          inet addr:192.168.1.221  Bcast:192.168.1.255  Mask:255.255.255.0
          inet6 addr: 2605:e000:5b0a:3e00:147a:fef:938:75a8/64 Scope:Global
          inet6 addr: fde9:7883:b343:0:3210:b3ff:fe40:e5/64 Scope:Global
          inet6 addr: fe80::3210:b3ff:fe40:e5/64 Scope:Link
          inet6 addr: fde9:7883:b343:0:147a:fef:938:75a8/64 Scope:Global
          inet6 addr: 2605:e000:5b0a:3e00:3210:b3ff:fe40:e5/64 Scope:Global
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:911018 errors:0 dropped:3 overruns:0 frame:0
          TX packets:318424 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:174958035 (166.8 MiB)  TX bytes:39902983 (38.0 MiB)

$ file -b script/work_flow.rb
Ruby script, ASCII text executable
$ file -b --mime script/work_flow.rb
text/x-ruby; charset=us-ascii
work_flow --<command> <file>
expected errors to be empty\ninspect="/home/greg/Desktop/src/Open-Table-Explorer/app/models/work_flow.rb:73:in `merge_range': :work_flow not found in [:passed, :testing, :edited] or {-2=>:\"origin/master\", -1=>:master} (RuntimeError)\n\tfrom /home/greg/Desktop/src/Open-Table-Explorer/app/models/work_flow.rb:289:in `merge_down'\n\tfrom script/work_flow.rb:114:in `block in <main>'\n\tfrom script/work_flow.rb:97:in `each'\n\tfrom script/work_flow.rb:97:in `<main>'\n" has no name. is not empty but contains "/home/greg/Desktop/src/Open-Table-Explorer/app/models/work_flow.rb:73:in `merge_range': :work_flow not found in [:passed, :testing, :edited] or {-2=>:\"origin/master\", -1=>:master} (RuntimeError)\n\tfrom /home/greg/Desktop/src/Open-Table-Explorer/app/models/work_flow.rb:289:in `merge_down'\n\tfrom script/work_flow.rb:114:in `block in <main>'\n\tfrom script/work_flow.rb:97:in `each'\n\tfrom script/work_flow.rb:97:in `<main>'\n"

  7) Error:
test_working_different_from?(WorkFlowTest):
RuntimeError: branch_index=nil
    /home/greg/Desktop/src/Open-Table-Explorer/app/models/work_flow.rb:45:in `branch_symbol?'
    /home/greg/Desktop/src/Open-Table-Explorer/app/models/work_flow.rb:110:in `diff_command?'
    test/unit/work_flow_test.rb:108:in `test_working_different_from?'
    /usr/lib/ruby/vendor_ruby/minitest/unit.rb:1058:in `run'
    /usr/lib/ruby/1.9.1/test/unit/testcase.rb:17:in `run'
    /usr/lib/ruby/vendor_ruby/active_support/testing/setup_and_teardown.rb:36:in `block in run'
    /usr/lib/ruby/vendor_ruby/active_support/callbacks.rb:403:in `_run__1874944238495754569__setup__1279465477025375313__callbacks'
    /usr/lib/ruby/vendor_ruby/active_support/callbacks.rb:405:in `__run_callback'
    /usr/lib/ruby/vendor_ruby/active_support/callbacks.rb:385:in `_run_setup_callbacks'
    /usr/lib/ruby/vendor_ruby/active_support/callbacks.rb:81:in `run_callbacks'
    /usr/lib/ruby/vendor_ruby/active_support/testing/setup_and_teardown.rb:35:in `run'
    /usr/lib/ruby/vendor_ruby/minitest/unit.rb:823:in `block in _run_suite'
    /usr/lib/ruby/vendor_ruby/minitest/unit.rb:815:in `map'
    /usr/lib/ruby/vendor_ruby/minitest/unit.rb:815:in `_run_suite'
    /usr/lib/ruby/1.9.1/test/unit.rb:565:in `block in _run_suites'
    /usr/lib/ruby/1.9.1/test/unit.rb:563:in `each'
    /usr/lib/ruby/1.9.1/test/unit.rb:563:in `_run_suites'
    /usr/lib/ruby/vendor_ruby/minitest/unit.rb:775:in `_run_anything'
    /usr/lib/ruby/vendor_ruby/minitest/unit.rb:964:in `run_tests'
    /usr/lib/ruby/vendor_ruby/minitest/unit.rb:951:in `block in _run'
    /usr/lib/ruby/vendor_ruby/minitest/unit.rb:950:in `each'
    /usr/lib/ruby/vendor_ruby/minitest/unit.rb:950:in `_run'
    /usr/lib/ruby/vendor_ruby/minitest/unit.rb:939:in `run'
    /usr/lib/ruby/1.9.1/test/unit.rb:21:in `run'
    /usr/lib/ruby/1.9.1/test/unit.rb:326:in `block (2 levels) in autorun'
    /usr/lib/ruby/1.9.1/test/unit.rb:27:in `run_once'
    /usr/lib/ruby/1.9.1/test/unit.rb:325:in `block in autorun'

33 tests, 660 assertions, 4 failures, 3 errors, 0 skips
